### DISCUSSION TITLE
Difference between 'active' and 'moving' in raw data?

### DISCUSSION THREAD

**Discussion by pazars** [SauceLLC/sauce4strava#145] [Q&A] [answered]
Maybe this is a bike computer specific question (I'm using a Wahoo Bolt V2), but when viewing the raw data, there are fields `active` and `moving`. What is the difference between the two and how do these relate to the actual moving time? 

To me they seem identical, but here's a little snippet of raw data from one of my rides that makes this a bit confusing.
```
time, active, moving, distance, watts, heartrate, cadence, velocity_smooth, altitude
...
603,   true,  false,   1709.7,   127,       143,      28,           1.134,      164
604,   true,  false,   1709.7,    95,       143,      16,           0.758,      164
605,  false,  false,   1709.9,     0,       144,       4,           0.456,      164
606,   true,   true,   1711.3,     0,       145,       0,           0.448,    164.2
607,   true,   true,   1712.8,     0,       145,       0,           0.606,    164.2
...
```

- You can see that at `time` 603 and 604 I was active but not moving, although there is power, cadence, and speed.
- At `time` 605 I was neither active nor moving, but there's still some cadence and speed. This would make sense if there is 
some lower speed/cadence threshold below which the data is ignored, however
- At `time` 606 I was moving and was active although speed and cadence was lower than at `time` 605.

**Comment by mayfield** [comment, marked as answer]
Hi @Pazars,

`Active` can be true when stationary if you have power or cadence data at the same stationary moments.  So for example sitting on a trainer or running in circles, etc.  In general `active` will be a little more inclusive and is used for more stats since it indicates actual time spent working.

EDIT:

I should add that `moving` is generated by Strava themselves, presumably using gps data for the most part and `active` is generated by Sauce using this function  https://github.com/SauceLLC/sauce4strava/blob/ffcf9dea5b4e1e891cb55bb7d4a6d8eedb44690d/src/common/lib.js#L124-L153

**Reply by pazars** [reply]
Thank you for the quick and clear answer!

Interesting that `sum(moving == true)` is not the same value as what Strava specifies as moving time for all rides. Makes the moving field a bit of a black box.

For me it's interesting to analyze rest time lengths, frequencies etc. and now it seems that using `active` is a better option.
